chain/chain/src/runtime/mod.rs

```diff
@@ -865,7 +866,7 @@ impl RuntimeAdapter for NightshadeRuntime {
                 storage_config.use_flat_storage,
             ),
         };
-        if storage_config.record_storage {
+        if checked_feature!("stable", StatelessValidationV0, PROTOCOL_VERSION) {
             trie = trie.recording_reads();
         }
```

Comment:
The recording should probably also be enabled with the `shadow_validation` feature.